src/main/cluster.ts

```diff
@@ -230,16 +230,12 @@ export class Cluster implements ClusterModel {
   }
 
   protected async k8sRequest<T = any>(path: string, options: RequestPromiseOptions = {}): Promise<T> {
+    (options.headers ??= {}).Host = `${this.id}.${new URL(this.kubeProxyUrl).host}` // required in ClusterManager.getClusterForRequest()
```

Comment:
It isn't optional, this line optionally initializes `options.headers` (just in case there weren't any passed in` and then overrides any `Host` field with that required value.